### DISCUSSION TITLE
Perfomance issues on zoom in VictoryLine

### DISCUSSION THREAD

**Discussion by vtarelkin** [FormidableLabs/victory#2302] [General]
Good day, trying to migrate from recharts and passing down the array of 6900 point of x and y like this
```


    const handleZoom = (domain) => {
        setSelectedDomain(domain);
    };
    const handleBrush = (domain) => {
        setZoomDomain(domain);
    };

    return (

            <VictoryChart
                width={550}
                height={300}
                containerComponent={
                    <VictoryZoomContainer
                        responsive={false}
                        zoomDimension="x"
                        zoomDomain={zoomDomain}
                        onZoomDomainChange={handleZoom}
                    />
                }
            >
                <VictoryLine
                    style={{
                        data: {stroke: "tomato"}
                    }}
                    data={data}
                    groupComponent={<g />}
                    x="x"
                    y="y0"
                />
            </VictoryChart>
        }
        </>
    )
```

and on zooming / moving everything is delayed. How can I remove it? 

![ezgif-1-24eb1bb981](https://user-images.githubusercontent.com/46703134/173825332-92d014d5-5782-40f0-9bce-9eaba93a5ebb.gif)

**Comment by pechea** [comment]
I experienced similar performance issue  with zoom.  after remove state change related code such as:
                        setSelectedDomain 
                        zoomDomain={zoomDomain}
                        onZoomDomainChange={handleZoom}
the performance got much better. zoom actually work without storing the zoom state externally
